### PR TITLE
Add azcopy_path in AzurePlatformSchema and use azcopy to copy blob

### PR DESCRIPTION
**Pull request by lubaihua33**

### CODE REVIEW

File: lisa/sut_orchestrator/azure/common.py [hunk 1]
```diff
@@ -1049,15 +1051,47 @@ def wait_copy_blob(
...
 
+def copy_blob_by_azcopy(
+    azcopy_path: str,
+    src: str,
+    dst: str,
+    log: Logger,
+    timeout: int = 60 * 60,
+) -> None:
+    try:
+        subprocess.check_output(
```
Comment (squirrelsc) [inline lisa/sut_orchestrator/azure/common.py]: use local node, so the log can be included in normal format.

File: lisa/sut_orchestrator/azure/common.py [hunk 2]
```diff
@@ -893,6 +894,7 @@ def generate_sas_token(
     account_name: str,
     resource_group_name: str,
     expired_hours: int = 2,
+    write: bool = False,
```
Comment (squirrelsc) [inline lisa/sut_orchestrator/azure/common.py]: `writable`

File: lisa/sut_orchestrator/azure/platform_.py [hunk 3]
```diff
@@ -1921,19 +1934,51 @@ def _get_deployable_vhd_path(
...
+                            "Then no need to check the hash key"
+                        )
+                        vhd_exists = True
 
             if not vhd_exists:
-                blob_client.start_copy_from_url(
-                    original_vhd_path, metadata=None, incremental_copy=False
-                )
+                azcopy_path = self._azure_runbook.azcopy_path
+                log.debug(f"AzCopy path:{azcopy_path}")
```
Comment (squirrelsc) [inline lisa/sut_orchestrator/azure/platform_.py]: This line should be printed, only azcopy_path specifies.

File: lisa/sut_orchestrator/azure/platform_.py [hunk 4]
```diff
@@ -1921,19 +1934,51 @@ def _get_deployable_vhd_path(
...
+                if azcopy_path and os.path.exists(azcopy_path):
```
Comment (squirrelsc) [inline lisa/sut_orchestrator/azure/platform_.py]: If the path specified, but not exists, it should raise an exception to make it obvious.

File: lisa/sut_orchestrator/azure/platform_.py [hunk 5]
```diff
@@ -1921,19 +1934,51 @@ def _get_deployable_vhd_path(
...
+                        )
+                    except Exception as identifier:
+                        blob_client.delete_blob(delete_snapshots="include")
+                        raise LisaException(f"{identifier}")
+
+                    # Set metadata to mark the blob copied by AzCopy successfully
+                    metadata = {"AzCopyStatus": "Success"}
+                    blob_client.set_blob_metadata(metadata)
+                    log.info("vhd copied")
+                    return full_vhd_path
```
Comment (squirrelsc) [inline lisa/sut_orchestrator/azure/platform_.py]: Why return here, not at the end?

File: lisa/sut_orchestrator/azure/platform_.py [hunk 6]
```diff
@@ -1921,19 +1934,51 @@ def _get_deployable_vhd_path(
...
+                else:
+                    blob_client.start_copy_from_url(
+                        src_vhd_sas_url, metadata=None, incremental_copy=False
+                    )
 
-            wait_copy_blob(blob_client, vhd_path, log)
+            wait_copy_blob(blob_client, dst_vhd_name, log)
```
Comment (squirrelsc) [inline lisa/sut_orchestrator/azure/platform_.py:1985]: If this line applies to Python SDK copy only, it should be in else part.

Comment (lubaihua33) [reply lisa/sut_orchestrator/azure/platform_.py:1985]: This line also applies to AzCopy when there is another copy operation in parallel.

File: lisa/sut_orchestrator/azure/platform_.py (hunk 5 above)
Comment (lubaihua33) [reply lisa/sut_orchestrator/azure/platform_.py]: The copy operation with AzCopy is synchronous. We don't need to call wait_copy_blob. So return here.
wait_copy_blob needs to be called when there is another copy operation in parallel.

File: lisa/sut_orchestrator/azure/platform_.py (hunk 4 above)
Comment (lubaihua33) [reply lisa/sut_orchestrator/azure/platform_.py]: Done

File: lisa/sut_orchestrator/azure/platform_.py (hunk 3 above)
Comment (lubaihua33) [reply lisa/sut_orchestrator/azure/platform_.py]: Have changed it to log.info().
I didn't get the point of  "only azcopy_path specifies." this comment. Change to log.info(f"{azcopy_path}") ?

File: lisa/sut_orchestrator/azure/common.py (hunk 2 above)
Comment (lubaihua33) [reply lisa/sut_orchestrator/azure/common.py]: Done

File: lisa/sut_orchestrator/azure/common.py (hunk 1 above)
Comment (lubaihua33) [reply lisa/sut_orchestrator/azure/common.py]: Used local().execute() and deleted the new added function copy_blob_by_azcopy

File: lisa/sut_orchestrator/azure/platform_.py (hunk 3 above)
Comment (squirrelsc) [reply lisa/sut_orchestrator/azure/platform_.py]: This log line should be printed, only the azcopy_path variable has value. If it's empty, it doesn't need to print empty string.

File: lisa/sut_orchestrator/azure/platform_.py (hunk 5 above)
Comment (squirrelsc) [reply lisa/sut_orchestrator/azure/platform_.py]: It still can reach the wait_copy_blob to make the logic consistent, and do an extra validation.

File: lisa/sut_orchestrator/azure/platform_.py (hunk 3 above)
Comment (lubaihua33) [reply lisa/sut_orchestrator/azure/platform_.py]: Done

File: lisa/sut_orchestrator/azure/platform_.py (hunk 5 above)
Comment (lubaihua33) [reply lisa/sut_orchestrator/azure/platform_.py]: Done

**Review by squirrelsc — APPROVED** [review]

**Comment by squirrelsc** [comment]
Please merge it next Monday after released.

**Pull request merged**